### ISSUE TITLE
Fix `no-invalid-position-declaration` false negatives for embedded blocks

### ISSUE THREAD

**Issue by jeddy3** [stylelint/stylelint#8719]
### What minimal example or steps are needed to reproduce the bug?

Run Stylelint against embedded blocks either in HTML or CSS-in-JS.

For example:

```html
<style>
  color: red;
</style>
```

Relates to https://github.com/stylelint/stylelint/pull/8697#issuecomment-3136936299, i.e. we intentionally favoured a false negative over a false positive.

### What minimal configuration is needed to reproduce the bug?

```json
{
  "customSyntax": "postcss-html",
  "rules": {
    "no-invalid-position-declaration": true
  }
}
```

### How did you run Stylelint?

[Demo](https://stylelint.io/demo/#N4Igxg9gJgpiBcIA8BnALgTwDYwHwB0A7AAmMiwgCd5jKYoBuIpAenWzxABoQAzASxwA5AIYBbOIhgAPcQAccAOgAWaMVm7gIhAQHMEIYEVL5wAV3QQxAZQyE0I6aZqm5EdGBQoAtKvWmuY2JTSjMcFGdiIxJSYJBCCG9+QgA3ESx+KG83FH40fm1vWDAsEUoRfO1ItFCYIIBfInrNSB1+XQAxKjEKgwArFG1NWDkUA2iTEHYcDPtI01K0GHQAoNNpmFm0b1a9b3QRQigyqHmQReW0VZjXdzRPHz8sM4uVkCaQeqA)

### Which Stylelint-related dependencies are you using?

```json
{
  "stylelint": "latest",
  "stylelint-config-standard": "latest",
  "postcss-html": "latest"
}
```

### What did you expect to happen?

`no-invalid-position-declaration` to flag the declaration.

### What actually happened?

The rule didn't flag the declaration.

### Do you have a proposal to fix the bug?

No, but the fix should be part of a broader initiative to resolve how we consistently handle custom syntaxes going forward, especially those with inline and block syntaxes like HTML:

```html
<a style="color: red;"></a>

<style>
  color: red;
</style>
```

See https://github.com/stylelint/stylelint/issues/6833

**Comment by sw1tch3roo** [comment]
working on this...

**Comment by MKroshko** [comment]
Have the same issue after migrating a project with styled-components to stylelint v16.
A simple repro demo could be found [with this link](https://stackblitz.com/edit/vitejs-vite-zhgcbkfb)
To see lint errors, stop the dev server and run `npm run lint` or `stylelint **/*.tsx --max-warnings 0` instead.
It should show:

```bush
src/App.tsx
   7:30  ✖  Unknown rule indentation                       indentation
   7:30  ✖  Unknown rule no-missing-end-of-source-newline  no-missing-end-of-source-newline
   8:3   ✖  Unexpected invalid position declaration        no-invalid-position-declaration
   9:3   ✖  Unexpected invalid position declaration        no-invalid-position-declaration
  10:3   ✖  Unexpected invalid position declaration        no-invalid-position-declaration
  11:3   ✖  Unexpected invalid position declaration        no-invalid-position-declaration
  13:3   ✖  Unexpected missing scoping root                nesting-selector-no-missing-scoping-root

✖ 7 problems (7 errors, 0 warnings)
```

Linter config file is as simple as:
```.stylelintrc.js
module.exports = {
  extends: [
    'stylelint-config-recommended',
    'stylelint-config-styled-components',
  ],
  rules: {
    indentation: 4,
    'selector-type-no-unknown': null,
  },
  overrides: [
    {
      files: ['**/*.{js,ts,jsx,tsx}'],
      customSyntax: '@stylelint/postcss-css-in-js',
    },
  ],
};
```

**Comment by MKroshko** [comment]
Ok, I found the way to fix errors like.
```
   8:3   ✖  Unexpected invalid position declaration        no-invalid-position-declaration
```
I was using the wrong customSyntax value. 
After changing `customSyntax: '@stylelint/postcss-css-in-js',` to `customSyntax: 'postcss-styled-syntax'` this type of errors gone for me.

Regarding all other errors (leaving this just for the history), my final config file looks like this:
```
module.exports = {
  plugins: ['@stylistic/stylelint-plugin'],
  extends: ['stylelint-config-standard'],
  rules: {
    '@stylistic/indentation': 2,
    'selector-type-no-unknown': null,
    'nesting-selector-no-missing-scoping-root': null,
    // Emulate package 'stylelint-config-styled-components' due to the bug https://github.com/stylelint/stylelint/issues/7669
    // ---------------start-------------------
    'value-no-vendor-prefix': true,
    'property-no-vendor-prefix': true,
    'no-empty-source': null,
    // "no-missing-end-of-source-newline": null
    // ---------------end---------------------
  },
  overrides: [
    {
      files: ['**/*.{js,ts,jsx,tsx}'],
      customSyntax: 'postcss-styled-syntax',
    },
  ],
};

```

**Comment by jeddy3** [comment]
Closed by https://github.com/stylelint/stylelint/pull/8907
